lib/plugins/s3/index.js

```diff
@@ -5,78 +5,75 @@ const s3 = require('s3');
 const Promise = require('bluebird');
 const throwIfMissing = require('../../support/util').throwIfMissing;
 const fs = require('fs-extra');
+const pick = require('lodash.pick');
 
 Promise.promisifyAll(fs);
 
-function getClient(options) {
-  if (options.s3.key && options.s3.secret) {
-    const settings = {
-      maxAsyncS3: options.s3.maxAsyncS3 || 20,
-      s3RetryCount: options.s3.s3RetryCount || 3,
-      s3RetryDelay: options.s3.s3RetryDelay || 1000,
-      multipartUploadThreshold: options.s3.multipartUploadThreshold ||  20971520,
-      multipartUploadSize: options.s3.multipartUploadSize || 15728640,
-      s3Options: {
-        accessKeyId: options.s3.key,
-        secretAccessKey: options.s3.secret
-      }
-    };
-    if (options.s3.region) {
-      settings.s3Options.region = options.s3.region;
-    }
+function createS3Client(s3Options) {
+  const clientOptions = pick(s3Options,
+    ['maxAsyncS3', 's3RetryCount', 's3RetryDelay', 'multipartUploadThreshold', 'multipartUploadSize']);
 
-    const client = s3.createClient(settings);
+  clientOptions.s3Options = {
+    accessKeyId: s3Options.key,
+    secretAccessKey: s3Options.secret,
+    region: s3Options.region
+  };
 
-    return client;
-  } else {
-    return s3.createClient();
-  }
+  return s3.createClient(clientOptions);
 }
 
 module.exports = {
   open(context, options) {
-    throwIfMissing(options.s3, ['bucketname'], 's3');
+    const s3Options = options.s3;
+
+    throwIfMissing(s3Options, ['bucketname'], 's3');
     if (options.outputFolder) {
       throw new Error('The current version does\'t support S3 together with a configured outputFolder.');
     }
-    if (options.s3.key || options.s3.secret) {
-      throwIfMissing(options.s3, ['key', 'secret'], 's3');
+    if (s3Options.key || s3Options.secret) {
+      throwIfMissing(s3Options, ['key', 'secret'], 's3');
```

Comment:
Is it really needed that both are passed as options? Or could one be passed as an option, and the other as an ENV variable? Perhaps not an interesting scenario in practise?